### PR TITLE
Publish npm package to Github and export http.Handler

### LINKED ISSUE #618: Add build pipeline steps for WeGO npm package

### PR DESCRIPTION
**Pull request by jpellizzari**
Closes #618

Splits some parts of the create an `npm` workflow and `pkg/server` package up to allow for other projects to consume it, per #600 

This PoC repo shows how to consume both the NPM package and Golang handler: https://github.com/jpellizzari/wego-wrapper-demo

NPM package: https://github.com/weaveworks/weave-gitops/packages/945027

### CODE REVIEW

**Comment by josecordaz** [comment]
I tried to run the demo example but got an issue trying to install dependencies. It seems I need permissions ? 🤔
image

**Review by josecordaz — APPROVED** [review]
LGTM

**Comment by jpellizzari** [comment]
> I tried to run the demo example but got an issue trying to install dependencies. It seems I need permissions ? 🤔
> image

@josecordaz Did you add an .npmrc file?

File: ui/index.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 <body>
     <div id="app"></div>
-    <script src="./main.js"></script>
+    <script type="module" src="./main.js"></script>
```
Comment (foot) [inline ui/index.html:13]: Out of curiosity does this change anything?

**Review by foot — APPROVED** [review]
I'm also seeing that auth error 🤔 , did you guys figure it out?

File: package.json [hunk 2]
```diff
@@ -13,54 +14,35 @@
...
   },
   "homepage": "https://github.com/weaveworks/weave-gitops#readme",
+  "peerDependencies": {
+    "lodash": "^4.17.21",
+    "luxon": "^1.27.0",
+    "react-dom": "^17.0.2",
+    "react-toastify": "^7.0.4",
+    "react": "^17.0.2",
+    "styled-components": "^5.3.0"
+  },
```
[the remark this answers is not in the document]
Comment (jpellizzari) [reply package.json:30]: Installing `@weaveworks/weave-gitops` will complain if you have different or missing deps in your project with a `WARN`. It just says to the consumer "hey this is what we need to run".

**Comment by jpellizzari** [comment]
Fixed in the reference repo by adding this to the `.npmrc` file:

```
//npm.pkg.github.com/:_authToken=${GITHUB_TOKEN}
```

The full `.npmrc` file needs to look like this:
```
@weaveworks:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken=${GITHUB_TOKEN}
```

**Review by luizbafilho — APPROVED** [review]

**Pull request merged**